src/ballet/txn/fd_txn.h

```diff
@@ -632,7 +642,57 @@ static inline ulong FD_FN_CONST fd_txn_acct_iter_end( void ) { return FD_TXN_ACC
    the parsing process will be accumulated into the struct pointed to by
    counters_opt. Note: The returned txn object is not self-contained
    since it refers to byte ranges inside the payload. */
-ulong fd_txn_parse( uchar const * payload, ulong payload_sz, void * out_buf, fd_txn_parse_counters_t * counters_opt );
+
+ulong
+fd_txn_parse_core( uchar const             * payload,
+                   ulong                     payload_sz,
+                   void                    * out_buf,
+                   fd_txn_parse_counters_t * counters_opt,
+                   ulong *                   payload_sz_opt,
+                   int allow_zero_signatures );
+
+static inline ulong fd_txn_parse( uchar const * payload, ulong payload_sz, void * out_buf, fd_txn_parse_counters_t * counters_opt ) {
+  return fd_txn_parse_core(payload, payload_sz, out_buf, counters_opt, NULL, 0);
+}
+
+struct fd_txn_xray_result {
+  uchar       signature_cnt;
+  ushort      signature_off;
+};
+typedef struct fd_txn_xray_result fd_txn_xray_result_t;
+
+ulong
+fd_txn_xray( uchar const             * payload,
+             ulong                     payload_sz,
+             fd_txn_xray_result_t    * result );
+
+static inline int
+fd_txn_is_writable( fd_txn_t const * txn, int idx ) {
+  if (txn->transaction_version == FD_TXN_V0 && idx >= txn->acct_addr_cnt) {
+    if (idx < (txn->acct_addr_cnt + txn->addr_table_adtl_writable_cnt)) {
+      return 1;
+    }
+    return 0;
+  }
+
+  if (idx < (txn->signature_cnt - txn->readonly_signed_cnt))
+    return 1;
+  if ((idx >= txn->signature_cnt) & (idx < (txn->acct_addr_cnt - txn->readonly_unsigned_cnt)))
+    return 1;
+
+
+  return 0;
+}
+
+static inline int
+fd_txn_is_signer( fd_txn_t const * txn, int idx ) {
+  return idx < txn->signature_cnt;
+}
+
+static inline ulong fd_txn_num_writable_accounts( fd_txn_t * txn ) {
+  return (ulong)((txn->signature_cnt - txn->readonly_signed_cnt)
+      + ((txn->acct_addr_cnt - txn->readonly_unsigned_cnt) - txn->signature_cnt));
+}
```

Comment:
You can replace this with the new `fd_txn_account_cnt` function which is more general and more precise. E.g. without looking at the implementation, I wouldn't know if this included writable accounts in the address lookup tables or not.

Reply:
removed